### PR TITLE
Add mnemonics scraper

### PR DESCRIPTION
**Pull request by renato262**
#### Ticket url(s):
- https://www.notion.so/c1841025a5424cc69b6a3ddc4a9e750a?v=b66a2adcc2aa4c76bd8315cebf00ab53&p=713cf3d755c64fd7b285b505a1c05852

#### Description:
_Add scrapy spiders to extract mnemonics data from websites._
- https://biolympiads.com/list-of-chemistry-mnemonics/
- https://en.wikipedia.org/wiki/List_of_mnemonics
- https://en.wikipedia.org/wiki/List_of_visual_mnemonics

#### Checklist:
- [x] Check that only one commit is added in the PR with the correct format
- [x] Rebase my branch

### CODE REVIEW

File: scraper/scraper/spiders/wikipedia_visuals.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                element_tag == "h2"
+                and element.xpath("span/text()").get() == self.exit_text
+            ):
+                return
+            if element_tag == "h3":
+                category_links = []
+                text = element.xpath(".//text()").get()
+                topic = text if text else topic
+            elif element_tag == "p" and topic:
+                item = MnemonicItem()
```
Comment (Drakko25) [inline scraper/scraper/spiders/wikipedia_visuals.py:52]: This `item` creation is being repeated many times, AFAICS, it the only attributes that are varying are `links`, `type` and `mnemonic`. So maybe we can move these parts to a method and make the code more readable.

File: scraper/scraper/spiders/wikipedia.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+                topic = text if text else topic
+
+            if pass_category:
+                continue
+            elif (
+                element_tag == "div" and element.xpath("@role").get() == "note"
+            ):  # get related topic url
+                category_links += element.xpath("a/@href").getall()
+
+            elif element_tag == "ul":  # get complex mnemonics or simple description
```
Comment (Drakko25) [inline scraper/scraper/spiders/wikipedia.py:91]: As the code is very big, it is hard to read and understand. Maybe we can put the code, for each `if`, in a separate method, like `process_ul_tag` if they won't require many params to be passed. If there are many shared variables, we can use something like a mixing to have those methods and inherit from it.

File: scraper/scraper/spiders/wikipedia_visuals.py (hunk 1 above)
Comment (renato262) [reply scraper/scraper/spiders/wikipedia_visuals.py]: Ok, I split in several methods

File: scraper/scraper/spiders/wikipedia.py (hunk 2 above)
Comment (renato262) [reply scraper/scraper/spiders/wikipedia.py:235]: Ok, I split in several methods

**Pull request merged**